float64 w1
float64 w2
